assessments/0010-etcd/etcd-implementation.md

```diff
@@ -0,0 +1,198 @@
+---
+title: Implementing etcd Doc Improvements
+tags: etcd
+---
+
+# Introduction
+
+This document provides actionable suggestions for improving the etcd technical documentaiton.  
+
+For an analysis and general discussion of recommendations on etcd technical documentation, see [etcd-analysis.md][etcd-analysis].
+
+## Recommendations, requirements, and best practices
+
+This analysis measures documentation against CNCF project maturity standards, and suggests possible improvements. In most cases there is more than one way to do things. Few recommendations here are meant to be prescriptive. Rather, recommendations are based on documentation best practices as understood by the reviewers. The recommended implementations represent the reviewers' experience with how to apply those best practices. In other words, borrowing terminology from the lexicon of [RFCs][rfc-keywords], the changes described here should be understood as "recommended" or "should" at the strongest, and "optional" or "may" in many cases. Any "must" or "required" actions are clearly denoted as such, and pertain to legal requirements such as copyright and licensing issues. 
+
+The top-level documentation recommendations for this project are:
+
+- Complete and update instructional documentation:
+    - Describe etcd's user roles (personas).
+    - Document Kubernetes and Linux installation procedures.
+    - Update and clarify quick start and new user workflows.
+    - Document key tasks as procedures rather than tutorials.
+    - Clarify which user roles (personas) each task is for.
+    - Going forward, write release notes for each major and minor release.
+- Reorganize the documentation.
+    - Move *project* documentation (documentation for OSS contributors) to the etcd GitHub repo. Move *product* documentation (documentation for those who use the etcd software, in any capacity) to the technical documentation in the website repo.
```

Comment:
What's the benefit of moving contributor documentation to the main repository?  I understand having it in its own section, but aren't there benefits to having it in the more readable & structured format supplied by doc publishing?

Reply:
Again, it's a best practice designed to separate content for different user roles. By analogy with commercial software, you wouldn't put (presumably internal) developer documentation in your user manuals. I realize that in OSS there's no proprietary information to keep hidden, but there are other reasons to maintain that separation. A big one in my experience is to avoid confusing new users, who don't yet have a good mental model of the product and are relying on the documentation to build that understanding.

That said, in OSS projects the lines are sometimes blurred between contributors and certain users. And there are sometimes good reasons to put contributor docs on the website; your readability example is one. I recommend that If you do put contributor documentation in the tech docs, clearly label it as such and isolate it from the user documentation.